src/libraries/System.IO.Pipes/src/System/IO/Pipes/ConnectionValueTaskSource.cs

```diff
@@ -1,17 +1,13 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
-using System.Runtime.ExceptionServices;
-using System.Threading;
-
 namespace System.IO.Pipes
 {
-    internal sealed class ConnectionCompletionSource : PipeCompletionSource<VoidResult>
+    internal sealed class ConnectionValueTaskSource : PipeValueTaskSource<VoidResult>
```

Comment:
I think we should preserve the old code and put it behind an app context switch, just like we did with the FileStream rewrite, so the user can fallback to that code just in case:

https://github.com/dotnet/runtime/blob/01b7e73cd378145264a7cb7a09365b41ed42b240/src/libraries/System.Private.CoreLib/src/System/IO/Strategies/FileStreamHelpers.cs#L10


@adamsitnik  @Jozkee  thoughts?

Reply:
I've not reviewed the change yet, but just based on PR title, that shouldn't be necessary here.  If someone is negatively impacted by using an IVTS here instead of a TCS, it's likely a bug in their code that should be fixed,  and such a change is on par with other bug fixes we make without a bug-for-bug compat fallback. 

I won't be able to review the change in detail until I'm back in a week,  though.